deepspeed/runtime/lr_schedules.py

```diff
@@ -381,6 +381,11 @@ def get_lr(self):
             lr_range_test_min_lr * lr_increase for lr_range_test_min_lr in self.min_lr
         ]
 
+    def get_last_lr(self):
+        """ Return last computed learning rate by current scheduler.
+        """
+        return self._last_lr
```

Comment:
Will this not fail with attribute error if the get_last_lr() is called right after object construction. I think self._last_lr needs to initialized in the constructor.

Reply:
while it shouldn't happen, technically yes, it'd fail, but to your suggestion - what would you assign to it as a default?

as I said I only mimicked how pytorch implemented it.

Probably the most clean implementation would be to assert inside `get_last_lr` if it's undefined. Should I do that?

Reply:
It seems to work in pytorch because LRScheduler, the base class, calls [step](https://github.com/pytorch/pytorch/blob/159f258415364f73a20ecd2b15b9f2e52ef1b922/torch/optim/lr_scheduler.py#L77) in the constructor, which sets self._last_lr. I don't think DeepSpeed lr schedulers are well setup to inherit from such a base class. So, I think an assert is fine to at least inform users of the failure reason.

Reply:
This is great. Thanks.